### ISSUE TITLE
ApplyConfigError: fiber is cancelled

### ISSUE THREAD

**Issue by rosik** [tarantool/cartridge#994]
Cartridge version: 2.2.0

Log:

```text
ApplyConfigError: fiber is cancelled
stack traceback:
        ...s/1.3.2-0/.rocks/share/tarantool/vshard/storage/init.lua:2428: in function 'cfg'
        ...rocks/share/tarantool/cartridge/roles/vshard-storage.lua:25: in function <...rocks/share/tarantool/cartridge/roles/vshard-storage.lua:11>
        [C]: in function 'xpcall'
        ...e-app/releases/1.3.2-0/.rocks/share/tarantool/errors.lua:148: in function 'apply_config'
        ...es/1.3.2-0/.rocks/share/tarantool/cartridge/failover.lua:246: in function <...es/1.3.2-0/.rocks/share/tarantool/cartridge/failover.lua:239>
        [C]: in function 'xpcall'
        ...e-app/releases/1.3.2-0/.rocks/share/tarantool/errors.lua:148: in function 'pcall'
        ...es/1.3.2-0/.rocks/share/tarantool/cartridge/failover.lua:239: in function <...es/1.3.2-0/.rocks/share/tarantool/cartridge/failover.lua:199>
```

**Comment by rosik** [comment]
### Bug origin

Here is approximate [code](https://github.com/tarantool/cartridge/blob/8502a9561a6622a4b83a9b6c997a982ccb81c5b4/cartridge/failover.lua#L199):

```lua 
local function failover_loop(args)
    while pcall(fiber.testcancel) do
        local appointments, err = FailoverError:pcall(args.get_appointments)
        if appointments == nil then
            -- don't log an error in case the fiber was cancelled
            fiber.testcancel()

            goto start_over
        end
        -- else 
        confapplier.set_state('ConfiguringRoles')
        -- configure roles
        confapplier.set_state('RolesConfigured')
end
```

It runs in a separate `failover_fiber` where `args.get_appointments` is

```lua
function()
    fiber.cond:wait()
    return {...}
end
```

The last piece of logic behind this bug is `failover.cfg` (as a part of two-phase commit), which does `failover_fiber:cancel()`

### Problem chronology

* Someone or something triggers two-phase commit which cancels `failover_fiber`.
* Instead of raising an error, `cond:wait()` returns `true` (see #5013) and failover keeps going.
* If there were some new appointments (it may be the case occasionally), it reaches `vshard.storage.cfg` internals
* vshard cancels some of his own fibers, but it fails because the calling fiber is already cancelled.
* The user observes "ApplyConfigError: fiber is cancelled".

**Comment by rosik** [comment]
By the way, the bug is already fixed by #937.

**Comment by rosik** [comment]
Released as [2.2.1](https://github.com/tarantool/cartridge/releases/tag/2.2.1)
